WordPress/Classes/Networking/MediaRequestAuthenticator.swift

```diff
@@ -0,0 +1,247 @@
+            provide(URLRequest(url: url))
+            return
+        }
+
+        let authenticationService = AtomicAuthenticationService(account: account)
+        let cookieJar = HTTPCookieStorage.shared
+
+        // Just in case, enforce HTTPs
+        components.scheme = "https"
+
+        guard let finalURL = components.url else {
+            fail(Error.cannotCreateAtomicURL(components: components))
+            return
+        }
+
+        let request = tokenAuthenticatedWPComRequest(for: finalURL, authToken: authToken)
+
+        authenticationService.loadAuthCookies(into: cookieJar, username: account.username, siteID: siteID, success: {
+            provide(request)
+        }) { error in
+            fail(Error.failedToLoadAtomicAuthenticationCookies(underlyingError: error))
+        }
+    }
+
+    /// Authentication for a Photon request in a private atomic site.
+    ///
+    /// - Important: Photon URLs are currently not working for private atomic sites, so this is a workaround
+    /// to replace those URLs with working URLs.
+    ///
+    /// By recommendation of @zieladam we'll be using the Atomic Proxy endpoint for these until
+    /// Photon starts working with Atomic Private Sites:
+    ///
+    /// https://public-api.wordpress.com/wpcom/v2/sites/$siteID/atomic-auth-proxy/file/$wpContentPath
+    ///
+    /// To know whether you can remove this method, try requesting the photon URL from an
+    /// atomic private site.  If it works then you can remove this workaround logic.
+    ///
+    /// - Parameters:
+    ///     - url: the url for the media.
+    ///     - siteID: the ID of the site that owns this media.
+    ///     - provide: the closure that will be called once authentication is sorted out by this class.
+    ///         The request can be executed directly without having to do anything else in terms of
+    ///         authentication.
+    ///     - fail: the closure that will be called upon finding an error condition.
+    ///
+    private func authenticatedRequestForPrivateAtomicSiteThroughPhoton(
+        for url: URL,
+        siteID: Int,
+        authToken: String,
+        onComplete provide: @escaping (URLRequest) -> (),
+        onFailure fail: @escaping (Error) -> ()) {
+
+        guard var components = URLComponents(url: url, resolvingAgainstBaseURL: true) else {
+            fail(Error.cannotBreakDownURLIntoComponents(url: url))
+            return
+        }
+
+        guard let wpContentRange = components.path.range(of: "/wp-content") else {
+            fail(Error.cannotFindWPContentInPhotonPath(components: components))
+            return
+        }
+
+        let contentPath = components.path[wpContentRange.lowerBound ..< components.path.endIndex]
+
+        components.scheme = "https"
+        components.host = "public-api.wordpress.com"
+        components.path = "/wpcom/v2/sites/\(siteID)/atomic-auth-proxy/file\(contentPath)"
```

Comment:
Should these fellows live in a constant (OR) specific method that builds the URL (for testing purposes)?

Reply:
Hey Jorge, I put the first two in constants since they're either used elsewhere or potentially reusable.

The third one, though... even if it'd make sense to put it inside a method of it's own, I find that readability suffers from it.  Additionally I'd want to discourage other methods from using the same path, unless there's a very good reason to.

As discussed privately I'll mark this as resolved.